File: razer-sdk/src/test/java/com/bonepl/razersdk/effects/StaticPartialEffectTest.java
package com.bonepl.razersdk.effects;

import com.bonepl.razersdk.ChromaRestSDK;
import com.bonepl.razersdk.animation.SimpleFrame;
import com.bonepl.razersdk.color.StaticColor;
import com.bonepl.razersdk.sdk.RzKey;
import org.junit.jupiter.api.Test;

import java.util.List;

import static com.bonepl.razersdk.sdk.RzKey.*;

class StaticPartialEffectTest {
    private final List<RzKey> smiley = List.of(RZKEY_W, RZKEY_3, RZKEY_4, RZKEY_R,
            RZKEY_I, RZKEY_9, RZKEY_0, RZKEY_P,
            RZKEY_X, RZKEY_SPACE, RZKEY_COMA);

    @Test
    @SuppressWarnings("JUnitTestMethodWithNoAssertions")
    void testPartialStaticEffect() throws InterruptedException {
        try (ChromaRestSDK razerSDKClient = new ChromaRestSDK()) {
            razerSDKClient.createKeyboardEffect(new SimpleFrame(smiley, StaticColor.YELLOW));
            Thread.sleep(100);
        }
    }
}

File: test/razer-sdk/effects/StaticPartialEffect.test.ts
import {describe, test} from 'vitest';
import {ChromaNativeSDK} from '../../../src/razer-sdk/ChromaNativeSDK.js';
import {SimpleFrame} from '../../../src/razer-sdk/animation/SimpleFrame.js';
import {StaticColor} from '../../../src/razer-sdk/color/StaticColor.js';
import {RzKey} from '../../../src/razer-sdk/sdk/RzKey.js';

const SMILEY: RzKey[] = [
    RzKey.RZKEY_W, RzKey.RZKEY_3, RzKey.RZKEY_4, RzKey.RZKEY_R,
    RzKey.RZKEY_I, RzKey.RZKEY_9, RzKey.RZKEY_0, RzKey.RZKEY_P,
    RzKey.RZKEY_X, RzKey.RZKEY_SPACE, RzKey.RZKEY_COMA,
];

/**
 * Integration test for partial keyboard lighting on actual hardware.
 * Draws a smiley face pattern using selected keys.
 */
describe('StaticPartialEffect Integration', () => {
    test('testPartialStaticEffect', async () => {
        const sdk = new ChromaNativeSDK();
        await sdk.connect();

        try {
            await sdk.createKeyboardEffect(new SimpleFrame(SMILEY, StaticColor.YELLOW));
            await new Promise(resolve => setTimeout(resolve, 2000));
        } finally {
            await sdk.close();
        }
    }, 30000);
});
